src/MarkedOptions.ts

```diff
@@ -150,3 +150,21 @@ export interface MarkedOptions extends Omit<MarkedExtension, 'hooks' | 'renderer
    */
   walkTokens?: null | ((token: Token) => void | Promise<void> | (void | Promise<void>)[]);
 }
+
+export interface MarkedSyncOptions extends _MarkedOptions {
+  async?: false;
+}
+
+export interface MarkedAsyncOptions extends _MarkedOptions {
+  async: true;
+}
+
+export type MarkedOptions = MarkedSyncOptions | MarkedAsyncOptions;
+
+export function isAsyncOptions(options: MarkedOptions): options is MarkedAsyncOptions {
+  return options.async === true;
+}
+
+export function isSyncOptions(options: MarkedOptions): options is MarkedSyncOptions {
+  return options.async === false;
```

Comment:
Should this be `false` or `undefined`?  Or maybe `!= true`?

Reply:
I pushed something like this:

```ts
return !isAsyncOptions(options);
```

It fail a lot of tests since it introduce a big breaking change.